apps/shared/PopupBlockedDialog.qml

```diff
@@ -0,0 +1,68 @@
+/****************************************************************************
+**
+** Copyright (c) 2021 Open Mobile Platform LLC.
+**
+****************************************************************************/
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import QtQuick 2.0
+import Sailfish.Silica 1.0
+import Sailfish.WebView.Controls 1.0
+
+Dialog {
+    id: dialog
+
+    property string host
+
+    property alias rememberValue: remember.checked
+
+    Column {
+        width: parent.width
+        spacing: Theme.paddingMedium
+
+        DialogHeader {
+            //: Allow the server to use load popup
+            //% "Allow"
+            acceptText: qsTrId("sailfish_browser-he-load_popup")
+
+            //: Deny the server to use load popup
+            //% "Deny"
+            cancelText: qsTrId("sailfish_browser-he-block_popup")
+        }
+
+        Label {
+            x: Theme.horizontalPageMargin
+            width: parent.width - x * 2
+
+            //: %1 is the site that wants load popup
+            //% "We have prevented %1 from opening pop-up window. Would you like to able this website open pop-ups?"
+            text: qsTrId("sailfish_browser-la-popup_request").arg(dialog.host)
+
+            wrapMode: Text.WordWrap
+            color: Theme.highlightColor
+        }
+
+        TextSwitch {
+            id: remember
+
+            //: Remember decision for this site for later use
+            //% "Remember for this site"
+            text: qsTrId("sailfish_browser-remember_for_site")
+        }
+    }
+    onAccepted: {
+        if (rememberValue) {
+            PermissionManager.add(dialog.host, "popup", PermissionManager.Allow)
+        } else {
+            PermissionManager.add(dialog.host, "popup", PermissionManager.Allow, PermissionManager.Session)
```

Comment:
One thing I find a little confusing is that the Permissions list doesn't distinguish between "Session allow" and "Always allow" permissions. Is this something that could or should be included?